cmd/log.go

```diff
@@ -4,21 +4,33 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"strings"
 	"time"
 
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+
 	log "github.com/sirupsen/logrus"
 )
 
-func init() {
+// initializeLoggerForConfig will configure the logger so that it matches the
+// user's specified configuration, such as configuring output files.
+func initializeLoggerForConfig(config runtime.Config) error {
 	time := time.Now().Unix()
-	logname := fmt.Sprintf("preflight-%d.log", time)
+	logname := strings.Join([]string{
+		config.CLILogFile, fmt.Sprint(time)}, "-")
+
 	logFile, err := os.OpenFile(logname, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0700)
 	if err == nil {
+		// if we could open the file for writing, write to both stdout and the file
+		// otherwise, just use stdout
 		mw := io.MultiWriter(os.Stdout, logFile)
 		log.SetOutput(mw)
 	} else {
-		log.Info("Failed to log to file, using default stderr")
+		log.Info("Failed to log to file, using default stdout")
```

Comment:
We can. For clarity, this change only made a change to the text output itself. We were already logging to stdout from what I could tell.

Caveat, here - if we log command line output (read: everything EXCEPT the results) to stderr, would we be violating the proper use of stderr? They technically aren't error lines (or I guess they could be).

I have another issue open to address writing results to a separate location as well: https://github.com/redhat-openshift-ecosystem/openshift-preflight/issues/14 if that helps with the comfort level of writing both to stdout.

Reply:
I don't think we would be violating it. It's pretty typical to separate log output to stderr (which might include actual errors) and the proper output to stdout. Think pipes... If we output json or xml, it would not be a stretch to send that output to a redirect or a pipe. If you put the log messages on stdout, you would no longer have the ability to send the output to jq for example.

Reply:
Understood - I'm with you now. This shoudl be fixed.